### PR TITLE
fix(TDI-43682): fix custom transaction types

### PR DESCRIPTION
**Pull request by mbasiuk-talend**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-43682

**What is the new behavior?**
Fix CustomTransaction Types.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [] The code coverage on new code >75%
- [x] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/255/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/255/)

**Comment by mbasiuk-talend** [comment]
```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Components - NetSuite Definition ................... SUCCESS [05:11 min]
[INFO] Components - NetSuite Runtime ...................... SUCCESS [02:35 min]
[INFO] Components - NetSuite Runtime v2019.2 .............. SUCCESS [01:01 min]
[INFO] Components - NetSuite Runtime v2018.2 .............. SUCCESS [ 54.647 s]
[INFO] Components - NetSuite Runtime v2016.2 .............. SUCCESS [ 49.787 s]
[INFO] Components - NetSuite Runtime v2014.2 .............. SUCCESS [ 41.173 s]
[INFO] Components - NetSuite Aggregator ................... SUCCESS [  0.618 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
```
Build actually passed.

File: components/components-netsuite/components-netsuite-runtime/src/main/java/org/talend/components/netsuite/client/DefaultMetaDataSource.java [hunk 1]
```diff
@@ -158,6 +159,10 @@ public TypeDesc getTypeInfo(final String typeName) {
                 CustomRecordTypeInfo customRecordTypeInfo = (CustomRecordTypeInfo) recordTypeInfo;
                 baseTypeDesc = clientService.getBasicMetaData().getTypeInfo(customRecordTypeInfo.getRecordType().getTypeName());
                 targetTypeName = customRecordTypeInfo.getName();
+            } else if (recordTypeInfo instanceof CustomTransactionTypeInfo) {
+                CustomTransactionTypeInfo customRecordTypeInfo = (CustomTransactionTypeInfo) recordTypeInfo;
+                baseTypeDesc = clientService.getBasicMetaData().getTypeInfo(customRecordTypeInfo.getRecordType().getTypeName());
+                targetTypeName = customRecordTypeInfo.getName();
```
Comment (gonchar-ivan) [inline components/components-netsuite/components-netsuite-runtime/src/main/java/org/talend/components/netsuite/client/DefaultMetaDataSource.java]: "if" and "else if" bodies looks the same for me. Am I missing something?
If they are the same, you may modify "if" condition in following way
```
if (recordTypeInfo instanceof CustomRecordTypeInfo || recordTypeInfo instanceof CustomTransactionTypeInfo) {
```
Also optionally conditiona may be extracted to separate method (but it's also ok to not extract the method)
```
private boolean isCustomType(RecordTypeInfo recordTypeInfo) {
    return recordTypeInfo instanceof CustomRecordTypeInfo || recordTypeInfo instanceof CustomTransactionTypeInfo;
```

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/257/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/257/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Ambasiuk%2FTDI-43682_disable_custom_fields_for_custom_transactions)

**Pull request merged**